win32ss/gdi/ntgdi/path.c

```diff
@@ -2106,15 +2099,18 @@ IntGdiWidenPath(PPATH pPath, UINT penWidth, UINT penStyle, FLOAT eMiterLimit)
         PATH_DestroyGdiPath(pDownPath);
         ExFreePoolWithTag(pDownPath, TAG_PATH);
     }
-    if (pStrokes) ExFreePoolWithTag(pStrokes, TAG_PATH);
 
-    PATH_UnlockPath(flat_path);
-    PATH_Delete(flat_path->BaseObject.hHmgr);
     pNewPath->state = PATH_Closed;
     PATH_UnlockPath(pNewPath);
 
     KeRestoreFloatingPointState(&fpsave);
 
+Exit:
+    if (pStrokes) ExFreePoolWithTag(pStrokes, TAG_PATH);
+    HPATH hpathToDelete = flat_path->BaseObject.hHmgr;
```

Comment:
Is the HPATH capture necessary because the PATH_UnlockPath call below may make `flat_path` invalid?

Reply:
Exactly. Eventually I'd like to have a better interface that makes such things unnecessary.